src/libostree/ostree-repo-private.h

```diff
@@ -413,8 +418,36 @@ _ostree_repo_get_remote_inherited (OstreeRepo  *self,
                                    const char  *name,
                                    GError     **error);
 
+gboolean
+_ostree_repo_delete_object_shared (OstreeRepo        *self,
+                                   OstreeObjectType   objtype,
+                                   const char        *sha256,
+                                   GCancellable      *cancellable,
+                                   GError           **error);
+
+
 #ifndef OSTREE_ENABLE_EXPERIMENTAL_API
 
+gboolean      ostree_repo_lock_push (OstreeRepo    *self,
+                                     gboolean       exclusive,
+                                     GCancellable  *cancellable,
+                                     GError       **error);
+gboolean      ostree_repo_lock_pop (OstreeRepo    *self,
+                                    GCancellable  *cancellable,
+                                    GError       **error);
+
+typedef OstreeRepo OstreeRepoAutoLock;
+OstreeRepoAutoLock * ostree_repo_auto_lock_push (OstreeRepo    *self,
+                                                 gboolean       exclusive,
```

Comment:
Might make sense to use an enum or flags instead of a boolean here, since it’s typically not obvious what the sense of a boolean is. Does `TRUE` mean ‘make it exclusive’ or ‘make it shared’?

Reply:
Yeah, I thought about that later after doing `TRUE`/`FALSE` so many times. It seemed sort of silly to have an enum that would only ever have 2 exclusive states. I guess:
```
typedef enum {
  OSTREE_REPO_LOCK_SHARED,
  OSTREE_REPO_LOCK_EXCLUSIVE
} OstreeRepoLockType;
```

Reply:
I think people (me included) should get out of the habit of thinking of two-element enums as ‘silly’, if they improve code readability. That said, there’s the possibility that this could be extended in future to include more flags or enum elements for more esoteric extensions to the locking.

Reply:
OK, I'll do it. That said, I would like this to remain a simple enum and not migrate into some flags setup later.